lib/logstash/outputs/elastic_app_search.rb

```diff
@@ -46,76 +42,61 @@ class LogStash::Outputs::ElasticAppSearch < LogStash::Outputs::Base
   # like `myapp-%{sequence_id}`. Reusing ids will cause documents to be rewritten.
   config :document_id, :validate => :string
 
-  # The path that is appended to the `url` parameter when connecting to a https://www.elastic.co/downloads/app-search
-  config :path, :validate => :string, :default => "/api/as/v1/"
+  ENGINE_WITH_SPRINTF_REGEX = /^.*%\{.+\}.*$/.freeze
 
-  ENGINE_WITH_SPRINTF_REGEX = /^.*%\{.+\}.*$/
-
-  public
   def register
-    @use_old_client = false
-    if @host.nil? && @url.nil?
-      raise ::LogStash::ConfigurationError.new("Please specify either \"url\" (for self-managed) or \"host\" (for SaaS).")
-    elsif @host && @url
-      raise ::LogStash::ConfigurationError.new("Both \"url\" or \"host\" can't be set simultaneously. Please specify either \"url\" (for self-managed ot Elastic Enterprise Search) or \"host\" (for SaaS).")
-    elsif @host && path_is_set?  # because path has a default value we need extra work to if the user set it
-      raise ::LogStash::ConfigurationError.new("The setting \"path\" is not compatible with \"host\". Use \"path\" only with \"url\".")
-    elsif @host
-      @deprecation_logger.deprecated("Deprecated service usage, the `host` setting will be removed when Swiftype AppSearch service is shutdown")
-      @use_old_client = true
-      @client = Elastic::AppSearch::Client.new(:host_identifier => @host, :api_key => @api_key.value)
-    elsif @url
-      if path_is_set?
-        @deprecation_logger.deprecated("Deprecated service usage, the `path` setting will be removed when Swiftype AppSearch service is shutdown")
-        @use_old_client = true
-        @client = Elastic::AppSearch::Client.new(:api_endpoint => @url + @path, :api_key => @api_key.value)
-      else
-        @client = Elastic::EnterpriseSearch::AppSearch::Client.new(:host => @url, :http_auth => @api_key.value, :external_url => @url)
-      end
-    end
-    check_connection! unless @engine =~ ENGINE_WITH_SPRINTF_REGEX
+    @retry_disabled = false
+    @client = LogStash::PluginMixins::EnterpriseSearch::AppSearch::Client.new(client_options, params: params)
+    check_connection!
   rescue => e
     if e.message =~ /401/
-      raise ::LogStash::ConfigurationError.new("Failed to connect to App Search. Error: 401. Please check your credentials")
+      raise LogStash::ConfigurationError, "Failed to connect to App Search. Please check your credentials. Error: #{e.message}"
```

Comment:
Why the absolute reference to the error was removed?
```ruby
::LogStash::ConfigurationError
```
is absolute reference, while

```ruby
LogStash::ConfigurationError
````
is partial. So if any plugin defines a class named for example `Plugin::Output::MyPlugin::LogStash::ConfigurationError` then that class could be selected instead of the Logstash's core provided.